### ISSUE TITLE
source: support native (data chunk) format for benchmark purpose

### ISSUE THREAD

**Issue by BugenZhao** [risingwavelabs/risingwave#4555]
When benchmarking with the Nexmark source (only with a `count(*)` simple agg), we find that even if we increase the split number (and the parallel unit count) from 6x3 to 7x3, the throughput does not improve while the CPU usage is increased by 100%. That is, the maximal throughput we can get from the Nexmark source is 750k rows per second; this will become the bottleneck for some workloads.

The Nexmark messages are generated and then serialized to JSON and then deserialized with the source parser, which brings some overhead. It would be better if we could support the "native" format by directly passing in-memory data chunks or columns.

<img width="1735" alt="image" src="https://user-images.githubusercontent.com/25862682/183823778-dedd0bc8-1a34-420c-b87a-6d728da55c20.png">
<img width="874" alt="image" src="https://user-images.githubusercontent.com/25862682/183823826-99c69597-7ad3-4996-b21f-b57949fd37e2.png">

**Comment by lmatz** [comment]
+1
check out
https://github.com/frankmcsherry/blog/blob/master/posts/2018-02-11.md
to see how people commit `Benchmarking Crimes`

**Comment by BugenZhao** [comment]
For example, 64% of the time of nexmark `bid` generation is spent on formatting the timestamp. However, there seems no way to optimize this like pre-compile the formatter. :( cc @KeXiangWang 

<img width="1038" alt="image" src="https://user-images.githubusercontent.com/25862682/183894310-36648a82-ebbe-4bca-b4dd-25b7f792c8ed.png">

**Comment by skyzh** [comment]
related: https://github.com/singularity-data/risingwave/issues/4144

**Comment by jon-chuang** [comment]
is there, equivalently, an observed high CPU consumption for deserializing string timestamp back to our in-memory representation of timestamp? (is that UTC time?). If so, then there would be a problem with _our_ code in being slow, not a problem with json serialization libraries as suggested in https://github.com/risingwavelabs/risingwave/issues/5122

**Comment by BugenZhao** [comment]
> is there, equivalently, an observed high CPU consumption for deserializing string timestamp back to our in-memory representation of timestamp? (is that UTC time as u64?). If so, then there would be a problem with _our_ code in being slow, not a problem with json serialization libraries as suggested in https://github.com/risingwavelabs/risingwave/issues/5122

I've not observed deserializing one. However, if we can avoid any json ser/de in the path of nexmark (actually no need at all), we can directly pass the mills u64 and it could be much better.

**Comment by fuyufjh** [comment]
This feature will facilitate the performance test of streaming operators, by eliminating the cost of Source Executor. Hope it can be done.

**Comment by fuyufjh** [comment]
@TennyZhuang Can you do it recently? Or reassign?

**Comment by fuyufjh** [comment]
> This feature will facilitate the performance test of streaming operators, by eliminating the cost of Source Executor. Hope it can be done.

To clarify, this feature is nice-to-have but not essential. We can do it as long as the complexity is controllable and does not hurt the current design of connectors.

**Comment by lmatz** [comment]
Just wondering if any design decision has been made, 
we can probably assign this to others if @wangrunji0408 is currently pretty busy with both external functions and expression micro-benchmarks/optimizations.

A quick check, I can only think of doing this unsafely without changing the code structure much(but still not sure if possible):
1. directly generate `StreamChunk` in `datagen` generator
2. unsafely reinterpret it as `[u8]`
3. a special parser in `ConnectorSourceReader` receives the `payload` (aka `[u8]`) and reconstructs the `StreamChunk`. 
Hope I am wrong!

**Comment by lmatz** [comment]
> make the throughput of data-gen larger than that of the downstream executor.

I see, yes! it is not particularly urgent. It's useful when we try to get some peak performance numbers 🤣 for demo purposes.

**Comment by wangrunji0408** [comment]
Hi all! Is this feature still required to meet performance requirements? If not, can I get it closed?

**Comment by fuyufjh** [comment]
> Hi all! Is this feature still required to meet performance requirements? If not, can I get it closed?

I think yes? cc. @lmatz 

BTW, this would be easier after https://github.com/risingwavelabs/rfcs/pull/31 cc. @waruto210

**Comment by waruto210** [comment]
> > Hi all! Is this feature still required to meet performance requirements? If not, can I get it closed?
> 
> I think yes? cc. @lmatz
> 
> BTW, this would be easier after [risingwavelabs/rfcs#31](https://github.com/risingwavelabs/rfcs/pull/31) cc. @waruto210

We can start doing it after #7508 is merged.

**Comment by waruto210** [comment]
done by #7621
